evals/llm.eval.ts

```diff
@@ -0,0 +1,206 @@
+import { Eval } from "braintrust";
+import { Levenshtein, ContextPrecision } from "autoevals";
+import goldResponses from "./data/eval-gold-responses.json";
+import OpenAI from "openai";
+import { VoyageAIClient } from "voyageai";
+import { createClient } from "@supabase/supabase-js";
+import { createAdapters } from "../src/adapters";
+import { Context } from "../src/types/context";
+import { logger } from "../src/helpers/errors";
+import { customOctokit as Octokit } from "@ubiquity-os/plugin-sdk/octokit";
+import issueTemplate from "../tests/__mocks__/issue-template";
+import { writeFileSync } from "fs";
+import { fetchContext, formattedHistory, initAdapters } from "./handlers/setup-context";
+
+// Required environment variables with type assertion
+const requiredEnvVars = {
+  OPENAI_API_KEY: process.env.OPENAI_API_KEY as string,
+  UBIQUITY_OS_APP_NAME: process.env.UBIQUITY_OS_APP_NAME as string,
+  VOYAGEAI_API_KEY: process.env.VOYAGEAI_API_KEY as string,
+  SUPABASE_URL: process.env.SUPABASE_URL as string,
+  SUPABASE_KEY: process.env.SUPABASE_KEY as string,
+  OPENROUTER_API_KEY: process.env.OPENROUTER_API_KEY as string,
+};
+
+// Validate all required env vars are present
+Object.entries(requiredEnvVars).forEach(([key, value]) => {
+  if (!value) {
+    throw new Error(`${key} is required`);
+  }
+});
+
+type Scenario = {
+  scenario: string;
+  issue: {
+    body: string;
+    html_url: string;
+    number: number;
+    question: string;
+  };
+  responseMustInclude: Array<string>;
+  sender: {
+    login: string;
+    type: string;
+  };
+  repository: {
+    name: string;
+    owner: {
+      login: string;
+      type: string;
+    };
+  };
+  expectedResponse: string;
+};
+
+type EvalInput = {
+  scenario: Scenario;
+};
+
+type EvalOutput = {
+  output: string;
+  context: string;
+  expected: string;
+};
+
+const inputs = {
+  config: {
+    model: "gpt-4o",
+    similarityThreshold: 0.8,
+    maxTokens: 1000,
```

Comment:
Why did you set such a low max? Also should we really be using GPT?

Reply:
There's also o1 now to consider using. 

To be frank I'm not super concerned with costs because we don't even use this feature every day but it can be tremendously valuable to automatically populate all the context and ask a question conveniently particularly from mobile which is my normal client. 

If it's providing high quality answers it's okay to spend a bit more in exchange for all the time savings.